schemas/2015-11-01/Microsoft.Network.json

```diff
@@ -0,0 +1,159 @@
+                                        ]
+                                    },
+                                    "properties": {
+                                        "type": "object",
+                                        "properties": {
+                                            "endpointStatus": {
+                                                "enum": [
+                                                    "Enabled",
+                                                    "Disabled"
+                                                ]
+                                            },
+                                            "targetResourceId": {
+                                                "type": "string"
+                                            },
+                                            "target": {
+                                                "type": "string"
+                                            },
+                                            "weight": {
+                                                "type": "integer",
+                                                "minimum": 1,
+                                                "maximum": 1000
+                                               },
+                                            "priority": {
+                                                "type": "integer",
+                                                "minimum": 1,
+                                                "maximum": 1000
+                                            },
+                                            "endpointLocation": {
+                                                "type": "string"
+                                            },
+                                            "minChildEndpoints": {
+                                                "type": "integer",
+                                                "minimum": 1
+                                            }
+                                        }
+                                    },
+                                    "required": [
+                                        "name",
+                                        "type",
+                                        "properties"
+                                    ]
+                                }
+                            },
+                            "additionalItems": true
+                        }
+                    },
+                    "required": [
+                        "trafficRoutingMethod",
+                        "dnsConfig",
+                        "monitorConfig"
+                    ],
+                    "additionalProperties": true
+                }
+            },
+            "required": [
+                "apiVersion",
+                "name",
+                "type",
+                "properties"
+            ],
+            "additionalProperties": true
```

Comment:
If template author set value on read-only properties in template, ARM service will send them in request body to RP, RP will ignore them. Wouldn't user be confused about why the value is not set successfully? Client side schema validation will mark the read-only properties as error so user can double check to understand it. Again, nothing blocks them to send the template containing resources with read-only properties to ARM, schema only do client side checking in VS during authoring template.